awx/main/analytics/collectors.py

```diff
@@ -174,7 +174,11 @@ def instance_info(since, include_hostnames=False):
             'memory': instance['memory'],
             'managed_by_policy': instance['managed_by_policy'],
             'last_isolated_check': _get_isolated_datetime(instance['last_isolated_check']),
-            'enabled': instance['enabled']
+            'enabled': instance['enabled'],
+            'consumed_capacity': sum(x.task_impact for x in models.UnifiedJob.objects.filter(execution_node=instance['hostname'],
```

Comment:
This has the potential to be **reeeeeeeeeally** expensive for databases that have large amounts of jobs (1M+).  One of the goals of this metric endpoint is to be as snappy as possible (given that people are polling it), so there's a good chance this would notably affect performance.

Reply:
@AlanCoding now that https://github.com/ansible/awx/pull/5472 has merged, I think this PR is probably much more reasonable.  What do you think?

Reply:
It's pretty self-obvious that it's still looping over the jobs. This will also probably trigger `O(N)` queries for the number of jobs, because `job.inventory` will incur an additional query just to get the inventory. This is hard to work around, because polymorphic. It still won't hit a major database performance bottleneck because of the tweak to total_hosts, yes.

But personally, I just don't have a good sense for what the _tolerance_ is for queries relative to the number of running jobs, for this particular collector code.

Reply:
How do we test this at a high-scale database under load? (hint hint)

Reply:
Yea, that's a fair point @AlanCoding.  I think this one is still probably something that needs to be in hiatus in its current state.